.clang-tidy

```diff
@@ -0,0 +1,6 @@
+---
+Checks:          '-*,google-*,cppcoreguidelines-*,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-pro-bounds-*,openmp-*,performance-*,portability-*,modernize-*,-modernize-use-trailing-*'
```

Comment:
This line removes all the default checks, and enables the following:
* Checks related to Google coding conventions
* Checks related to C++ Core Guidelines except avoid magic numbers and [bounds checks](https://releases.llvm.org/3.8.0/tools/clang/tools/extra/docs/clang-tidy/checks/cppcoreguidelines-pro-bounds-constant-array-index.html) 
* OpenMP APIs
* performance related issues
* portability related issues
* promote modern C++11 language constructs except forcing a [trailing return type](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-trailing-return-type.html)